pkg/services/ngalert/models.go

```diff
@@ -0,0 +1,92 @@
+package ngalert
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/eval"
+)
+
+// AlertDefinition is the model for alert definitions in Alerting NG.
+type AlertDefinition struct {
+	Id        int64
+	OrgId     int64
+	Name      string
+	Condition string
+	Data      []AlertQuery
+}
+
+var (
+	// ErrAlertDefinitionNotFound is an error for an unknown alert definition.
+	ErrAlertDefinitionNotFound = fmt.Errorf("could not find alert definition")
+)
+
+// GetAlertDefinitionByIDQuery is the query for retrieving/deleting an alert definition by ID.
+type GetAlertDefinitionByIDQuery struct {
+	ID    int64
+	OrgID int64
+
+	Result *AlertDefinition
+}
+
+type DeleteAlertDefinitionByIDQuery struct {
+	ID    int64
+	OrgID int64
+
+	RowsAffected int64
+}
+
+// Condition is the structure used by storing/updating alert definition commmands
+type Condition struct {
```

Comment:
This doesn't need exporting.
```suggestion
// condition is used for storing/updating alert definitions.
type condition struct {
```